File: packages/sdk-react-native/ios/EthereumRequestObjC.m
#import "EthereumRequestObjC.h"
#import <React/RCTLog.h>

@implementation EthereumRequestObjC

@synthesize method = _method;
@synthesize params = _params;

- (instancetype)initWithDictionary:(NSDictionary *)dictionary {
  if (self = [super init]) {
    _method = dictionary[@"method"];
    _params = dictionary[@"params"];
  }
  return self;
}

@end

File: packages/sdk-react-native/ios/EthereumRequestObjC.swift
//
//  EthereumRequestObjC.swift
//  MetaMaskNativeSDK
//

import Foundation
import metamask_ios_sdk

@objc class EthereumRequestObjC: NSObject {
  var method: String
  var params: Any

  @objc init(method: String, params: Any) {
      self.method = method
      self.params = params
  }
}
